test/test_images/probe_helper/probe/helper.go

```diff
@@ -0,0 +1,327 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/storage"
+	cepubsub "github.com/cloudevents/sdk-go/protocol/pubsub/v2"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+	"go.uber.org/zap"
+
+	"knative.dev/pkg/logging"
+
+	schemasv1 "github.com/google/knative-gcp/pkg/schemas/v1"
+	"github.com/google/knative-gcp/test/test_images/probe_helper/utils"
+)
+
+const (
+	probeEventTimeoutExtension       = "timeout"
+	probeEventTargetServiceExtension = "targetservice"
```

Comment:
Just came to my mind - if we are going to use URI to target the event consumer, then we don't have the "same namespace" constraint any more. Hence instead of creating a new ExternalName service, we can just use a path in the probe helper receiver, like so: <probe_helper_receiver_uri>/<namespace>/<trigger>. So by parsing this path we know exactly the expected trigger name and its namespace.

Reply:
Good point! Maybe the extension carrying this information can become "targetpath" instead then?

Reply:
We are already using "namespace" and "broker" for the e2e delivery case. I wonder if we can just do similarly: just use "namespace" and "trigger".

Reply:
Settled on targetpath.